src/signer_node/node_state.rs

```diff
@@ -31,6 +31,7 @@ pub enum NodeState {
         participants: HashSet<SignerID>,
         /// Set true when the round is done.
         round_is_done: bool,
+        block_height: u64,
```

Comment:
How about make the NodeParameters have block_height instead of NodeState variants?

It can eliminate `block_height` argument from NodeParameters methods.

Reply:
block_height is updated for each block.
I think NodeParameters is designed to be immutable through the rounds.